Big_Integer/main.cpp

```diff
@@ -0,0 +1,476 @@
+        string diff = "";
+        int i = (int)a.size() - 1;
+        int j = (int)b.size() - 1;
+        int k = 0;
+        while (i >= 0) {
+            int s = a[i] - '0' - k;
+            if (j >= 0) {
+                s =s-( b[j] - '0');
+            }
+            if (s < 0) {
+                s += 10;
+                k = 1;
+            } else {
+                k = 0;
+            }
+            i --;
+            j --;
+            diff =diff + (char)(s + '0');
+        }
+        while ((int)diff.size() > 1 && diff.back() == '0') {
+            diff.pop_back();
+        }
+        reverse(diff.begin(), diff.end());
+        return diff;
+    }
+    void _break(string &a, string &l, string &r) { // break the number two 2 equal parts
+        int length = (int)a.size();
+        int half = length >> 1;
+        l = a.substr(0, half);
+        r = a.substr(half);
+    }
+    void normalize(string &a, string &b) { // used to make the len equal to two's power
+        int length_A = (int)a.size();
+        int length_B = (int)b.size();
+        int to = -1;
+        for (int i = 0; ; i ++) {
+            if ((1 << i) >= length_A && (1 << i) >= length_B) {
+                to = (1 << i);
+                break;
+            }
+        }
+        if (length_A < to) {
+            reverse(a.begin(), a.end());
+            while (length_A < to) {
+                a += '0';
+                length_A ++;
+            }
+            reverse(a.begin(), a.end());
+        }
+        if (length_B < to) {
+            reverse(b.begin(), b.end());
+            while (length_B < to) {
+                b += '0';
+                length_B ++;
+            }
+            reverse(b.begin(), b.end());
+        }
+
+    }
+    string defaultMultiplier(string &a, string &b) { // builtin multiplication
+        int aValue = stoi(a); //string to int
+        int bValue = stoi(b);
+        return to_string(1LL * aValue * bValue);
+    }
+
+    string multiply(string a, string b) {     //Day 5: Using karatsuba Algorithm...
+        if ((int)a.size() < 9 && (int)b.size() < 9) {
+            return defaultMultiplier(a, b);
+        }
+
+        normalize(a, b);
+        string leftA, rightA;
+        string leftB, rightB;
+        _break(a, leftA, rightA);
+        _break(b, leftB, rightB);
+        string p1 = multiply(leftA, leftB);
+        string p2 = multiply(rightA, rightB);
+        string a_sum = add(leftA, rightA);
+        string b_sum = add(leftB, rightB);
+        string p3 = multiply(a_sum, b_sum);
+        string delta31 = subtract(p3, p1);
+        string p4 = subtract(delta31, p2);
+        insertTrailingZeros(p1, (int)a.size());
+        insertTrailingZeros(p4, (int)a.size() >> 1);
+        string sum= add(p2, p4);
+        return add(p1, sum);
+    }
+
+
+
+    string _modulo(string a, string b) { // modulo operation of two integers(strings)
+        string result = "";
```

Comment:
Codacy found an issue: [Variable 'result' is assigned a value that is never used.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290669)